fe/src/main/java/org/apache/doris/catalog/OlapTable.java

```diff
@@ -1115,4 +1115,20 @@ public int getKeysNum() {
         }
         return keysNum;
     }
+
+    public boolean convertRandomDistributionToHashDistribution() {
+        boolean hasChanged = false;
+        List<Column> baseSchema = indexIdToSchema.get(baseIndexId);
+        if (defaultDistributionInfo.getType() == DistributionInfoType.RANDOM) {
+            defaultDistributionInfo = ((RandomDistributionInfo) defaultDistributionInfo).toHashDistributionInfo(baseSchema);
+            hasChanged = true;
+        }
+        
+        for (Partition partition : idToPartition.values()) {
+            if (partition.convertRandomDistributionToHashDistribution(baseSchema)) {
```

Comment:
In which case defaultDistributionInfo is random, however the partition is not random partition?

Reply:
I don't know, just for safety...

Reply:
OK